enterprise_access/apps/api/v1/views/assignment_configuration.py

```diff
@@ -0,0 +1,253 @@
+        given ``SubsidyAccessPolicy`` uuid.
+        """
+        response = super().create(request, *args, **kwargs)
+        if not self.extra_context.get('created'):
+            response.status_code = status.HTTP_200_OK
+        return response
+
+    @extend_schema(
+        tags=[CONTENT_ASSIGNMENTS_CONFIGURATION_CRUD_API_TAG],
+        summary='Partially update (with a PUT) a content assignment configuration by UUID.',
+        request=serializers.AssignmentConfigurationUpdateRequestSerializer,
+        responses={
+            status.HTTP_200_OK: serializers.AssignmentConfigurationResponseSerializer,
+            status.HTTP_404_NOT_FOUND: None,
+        },
+    )
+    @permission_required(CONTENT_ASSIGNMENTS_CONFIGURATION_WRITE_PERMISSION, fn=assignment_config_permission_detail_fn)
+    def update(self, request, *args, uuid=None, **kwargs):
+        """
+        Updates a single ``AssignmentConfiguration`` record by uuid.  All fields for the update are optional (which is
+        different from a standard PUT request).
+        """
+        kwargs['partial'] = True
+        return super().update(request, *args, uuid=uuid, **kwargs)
+
+    @extend_schema(
+        tags=[CONTENT_ASSIGNMENTS_CONFIGURATION_CRUD_API_TAG],
+        summary='Partially update (with a PATCH) a content assignment configuration by UUID.',
+        request=serializers.AssignmentConfigurationUpdateRequestSerializer,
+        responses={
+            status.HTTP_200_OK: serializers.AssignmentConfigurationResponseSerializer,
+            status.HTTP_404_NOT_FOUND: None,
+        },
+    )
+    @permission_required(CONTENT_ASSIGNMENTS_CONFIGURATION_WRITE_PERMISSION, fn=assignment_config_permission_detail_fn)
+    def partial_update(self, request, *args, uuid=None, **kwargs):
+        """
+        Updates a single ``AssignmentConfiguration`` record by uuid.  All fields for the update are optional.
+        """
+        return super().partial_update(request, *args, uuid=uuid, **kwargs)
+
+    @extend_schema(
+        tags=[CONTENT_ASSIGNMENTS_CONFIGURATION_CRUD_API_TAG],
+        summary='Soft-delete content assignment configuration by UUID.',
+        request=serializers.AssignmentConfigurationDeleteRequestSerializer,
+        responses={
+            status.HTTP_200_OK: serializers.AssignmentConfigurationResponseSerializer,
+            status.HTTP_404_NOT_FOUND: None,
+        },
+    )
+    @permission_required(CONTENT_ASSIGNMENTS_CONFIGURATION_WRITE_PERMISSION, fn=assignment_config_permission_detail_fn)
+    def destroy(self, request, *args, uuid=None, **kwargs):
+        """
+        Soft-delete a single ``AssignmentConfiguration`` record by uuid, and unlink from the associated policy.
+
+        Note: This endpoint supports an optional "reason" request body parameter, representing the description (free
+        form text) for why the AssignmentConfiguration is being deactivated.
```

Comment:
This is a sad fact.  I can't get the "reason" body parameter to show up in redoc or swagger.  No idea why drf-spectacular won't pick it up from the serializer, and I also can't figure out how to override the schema in a sane way.  Not worth the battle IMO.

Reply:
could be related to DELETE methods not officially supporting request bodies according to IETF RFC specs.